### PR TITLE
Add support for _ in bytepattern captures

### PR DESCRIPTION
**Pull request by wesleimp**
Fixes: #51 

@leostera I've added some test for that scenario, but all tests passed. Am I doing this in a wrong way, or maybe this issue was already fixed?

### CODE REVIEW

**Comment by leostera** [comment]
Hi @wesleimp, thanks for opening this! ✨ Looks good to me! Can you add a test to `bytestring/ppx.t/matching.ml` ? We need a pattern that uses an underscore in the captured variable name. If that passes then maybe we'll need to revisit the issue and see if it is no longer valid.

File: bytestring/ppx.t/matching.ml [hunk 1]
```diff
@@ -33,7 +33,7 @@ let () =
         fin + comp
     | {| fin::1, comp::1, 0::2, 1::4, 0::1, 127::7, |} -> fin + comp
     | {| fin::1, comp::1 |} -> fin + comp
-    | {| rest |} -> Bytestring.length rest
+    | {| the_rest |} -> Bytestring.length the_rest
```
Comment (wesleimp) [inline bytestring/ppx.t/matching.ml:36]: Capturing and using the captured variable

File: bytestring/ppx.t/matching.ml [hunk 2]
```diff
@@ -70,10 +70,10 @@ let () =
        length::7,
        mask::32,
        _payload::bytes(length * 8),
-       _rest::bytes |}
+       _rest_rest::bytes |}
```
Comment (wesleimp) [inline bytestring/ppx.t/matching.ml:73]: Capturing but not using it

**Comment by leostera** [comment]
Hmm. One last thing we can try is a pattern match where the only thing we do is 

```ocaml
match%b data with
| {| hello_world |} -> ...
```

so we capture the whole thing under a variable with underscores.

Sorry to bug you with all these tiny tests, its just that my gut just tells me there's a test there that is going to break. I may be wrong.

If that passes, then we can merge your tests and close the issue 🙏🏼

**Comment by leostera** [comment]
Thanks @wesleimp! 🙌🏼

**Pull request merged**